### DISCUSSION TITLE
Install problem

### DISCUSSION THREAD

**Discussion by BruceLee0718** [inducer/pycuda#447] [Troubleshooting]
**Describe the bug**
When I am installing the pycuda, I got the following errors:
   Building wheel for pycuda (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pycuda (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [153 lines of output]
      *************************************************************
      *** I have detected that you have not run configure.py.
      *************************************************************
      *** Additionally, no global config files were found.
      *** I will go ahead with the default configuration.
      *** In all likelihood, this will not work out.
      ***
      *** See README_SETUP.txt for more information.
      ***
      *** If the build does fail, just re-run configure.py with the
      *** correct arguments, and then retry. Good luck!
      *************************************************************
      *** HIT Ctrl-C NOW IF THIS IS NOT WHAT YOU WANT
      *************************************************************
      Continuing in 10 seconds...
      Continuing in 9 seconds...
      Continuing in 8 seconds...
      Continuing in 7 seconds...
      Continuing in 6 seconds...
      Continuing in 5 seconds...
      Continuing in 4 seconds...
      Continuing in 3 seconds...
      Continuing in 2 seconds...
      Continuing in 1 seconds...
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-37
      creating build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/_mymako.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/autoinit.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/driver.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/elementwise.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/debug.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/curandom.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/__init__.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/autoprimaryctx.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/_cluda.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/compiler.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/reduction.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/scan.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/characterize.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/gpuarray.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/cumath.py -> build/lib.linux-x86_64-cpython-37/pycuda
      copying pycuda/tools.py -> build/lib.linux-x86_64-cpython-37/pycuda
      creating build/lib.linux-x86_64-cpython-37/pycuda/gl
      copying pycuda/gl/autoinit.py -> build/lib.linux-x86_64-cpython-37/pycuda/gl
      copying pycuda/gl/__init__.py -> build/lib.linux-x86_64-cpython-37/pycuda/gl
      creating build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/packeted.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/pkt_build.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/cg.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/coordinate.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/__init__.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/inner.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      copying pycuda/sparse/operator.py -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      creating build/lib.linux-x86_64-cpython-37/pycuda/compyte
      copying pycuda/compyte/__init__.py -> build/lib.linux-x86_64-cpython-37/pycuda/compyte
      copying pycuda/compyte/array.py -> build/lib.linux-x86_64-cpython-37/pycuda/compyte
      copying pycuda/compyte/dtypes.py -> build/lib.linux-x86_64-cpython-37/pycuda/compyte
      running egg_info
      writing pycuda.egg-info/PKG-INFO
      writing dependency_links to pycuda.egg-info/dependency_links.txt
      writing requirements to pycuda.egg-info/requires.txt
      writing top-level names to pycuda.egg-info/top_level.txt
      reading manifest file 'pycuda.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      /tmp/pip-build-env-ad4u47ee/overlay/lib/python3.7/site-packages/setuptools/_distutils/dist.py:265: UserWarning: Unknown distribution option: 'test_requires'
        warnings.warn(msg)
      warning: no files found matching 'doc/source/*.rst'
      warning: no files found matching 'doc/source/conf.py'
      warning: no files found matching 'doc/source/_static/*.css'
      warning: no files found matching 'doc/source/_templates/*.html'
      warning: no files found matching '*.cpp' under directory 'bpl-subset/bpl_subset/boost'
      warning: no files found matching '*.html' under directory 'bpl-subset/bpl_subset/boost'
      warning: no files found matching '*.inl' under directory 'bpl-subset/bpl_subset/boost'
      warning: no files found matching '*.txt' under directory 'bpl-subset/bpl_subset/boost'
      warning: no files found matching '*.h' under directory 'bpl-subset/bpl_subset/libs'
      warning: no files found matching '*.ipp' under directory 'bpl-subset/bpl_subset/libs'
      warning: no files found matching '*.pl' under directory 'bpl-subset/bpl_subset/libs'
      adding license file 'LICENSE'
      writing manifest file 'pycuda.egg-info/SOURCES.txt'
      creating build/lib.linux-x86_64-cpython-37/pycuda/cuda
      copying pycuda/cuda/pycuda-complex-impl.hpp -> build/lib.linux-x86_64-cpython-37/pycuda/cuda
      copying pycuda/cuda/pycuda-complex.hpp -> build/lib.linux-x86_64-cpython-37/pycuda/cuda
      copying pycuda/cuda/pycuda-helpers.hpp -> build/lib.linux-x86_64-cpython-37/pycuda/cuda
      copying pycuda/sparse/pkt_build_cython.pyx -> build/lib.linux-x86_64-cpython-37/pycuda/sparse
      running build_ext
      building '_driver' extension
      creating build/temp.linux-x86_64-cpython-37
      creating build/temp.linux-x86_64-cpython-37/bpl-subset
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/python
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/python/src
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/python/src/converter
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/python/src/object
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/smart_ptr
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/smart_ptr/src
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/system
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/system/src
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/thread
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/thread/src
      creating build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/thread/src/pthread
      creating build/temp.linux-x86_64-cpython-37/src
      creating build/temp.linux-x86_64-cpython-37/src/cpp
      creating build/temp.linux-x86_64-cpython-37/src/wrapper
      gcc -pthread -B /home/server/anaconda3/envs/GLAD/compiler_compat -Wl,--sysroot=/ -Wsign-compare -fwrapv -Wall -O3 -DNDEBUG -fPIC -DBOOST_ALL_NO_LIB=1 -DBOOST_THREAD_BUILD_DLL=1 -DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION=1 -DBOOST_PYTHON_SOURCE=1 -Dboost=pycudaboost -DBOOST_THREAD_DONT_USE_CHRONO=1 -DPYGPU_PACKAGE=pycuda -DPYGPU_PYCUDA=1 -DHAVE_CURAND=1 -Isrc/cpp -Ibpl-subset/bpl_subset -I/usr/local/cuda/include -I/tmp/pip-build-env-ad4u47ee/overlay/lib/python3.7/site-packages/numpy/core/include -I/home/server/anaconda3/envs/GLAD/include/python3.7m -c bpl-subset/bpl_subset/libs/python/src/converter/arg_to_python_base.cpp -o build/temp.linux-x86_64-cpython-37/bpl-subset/bpl_subset/libs/python/src/converter/arg_to_python_base.o
      cc1plus: error: unrecognized command-line option ‘-auxbase-strip’
      cc1plus: error: too many filenames given; type ‘cc1plus --help’ for usage
      /tmp/pip-build-env-ad4u47ee/overlay/lib/python3.7/site-packages/setuptools/command/build_py.py:201: _Warning: Package 'pycuda.cuda' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pycuda.cuda' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pycuda.cuda' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pycuda.cuda' to be distributed and are
              already explicitly excluding 'pycuda.cuda' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycuda
Failed to build pycuda
ERROR: Could not build wheels for pycuda, which is required to install pyproject.toml-based projects

**To Reproduce**

**Expected behavior**

**Environment (please complete the following information):**
 - OS:Ubuntu22.04 server [e.g. iOS]
 - CUDA version: 11.1 [e.g. 11.1]
 - CUDA driver version:535.171.04 [e.g. 470.86]
  - PyCUDA version: 2021.1 [e.g 2021.1]
  - Python version:3.7  [e.g. 3.10]

**Additional context**

**Comment by inducer** [comment]
Please test that your `gcc` can successfully compile C++ code.
